### ISSUE TITLE
Include Videos (and other mime types?) in selection

### ISSUE THREAD

**Issue by calebisstupid** [Sh1d0w/multi_image_picker#69]
Request: Allow selection of videos (and other media types) supported by Matisse and BSImagePicker.

Matisse supports video selection already. Today I put together solution in our fork to get video selection working on Android. It is not clean but more of a proof of concept.

I am planning to do a PR for BSImagePicker (ideally this week) to get video selection enabled there. Brief response from that repo's moderator says it should be merged if it looks clean.

**Comment by Sh1d0w** [comment]
Hi and thanks for the suggestion.

This is a big feature and probably will require a lot of changes. I don't have the time currently, but PRs are always welcome.

**Comment by calebisstupid** [comment]
Sounds good. If I get the iOS side working soon, I plan to circle back to the Android piece and actually make it usable and not just some proof of concept.

I currently have the android side loading video thumbnails and video data, but just gotta clean up some code and add some additional options for customizable mime types (so in Flutter you could do something like "images" "videos" or "both"). I'll definitely submit a PR if I get it cleaned up like this.

**Comment by calebisstupid** [comment]
@Sh1d0w So I have BSImagePicker near ready for a PR for videos, and I'm moving on to working on modifications for this plugin to clean up my work from yesterday and hopefully get a PR ready. For the life of me, I cannot find out how to reference a local pod directory (my copy of BSImagePicker) from within a Flutter package. Do you have any experience with this?

I've modified the podspec file to use the local directory and I've added a line in the example project podfile to use the local, all pod logs indicate its pulling my local copy but then when it all compiles it's using the original 2.9 version :-(

**Comment by calebisstupid** [comment]
Just got it. Ended up keeping podspec the same, and making sure the example project podfile referenced my local path before anything else. The key was running a pod clean then another pod install to make sure it was pulling from my local.

**Comment by calebisstupid** [comment]
For reference: my PR to BSImagePicker (CI build is broken, so I'm trying to figure out what happened there :/) https://github.com/mikaoj/BSImagePicker/pull/197

**Comment by Sh1d0w** [comment]
@calebisstupid Nice work! Glad you've figured it out. 👌

**Comment by calebisstupid** [comment]
@Sh1d0w looks like BSImagePicker may be getting video support soon in a new 3.0 version (https://github.com/mikaoj/BSImagePicker/pull/197)

**Comment by Sh1d0w** [comment]
@calebisstupid Thanks for your effort. However recently I released version 4 of multi_image_picker, which dropped support for the Android library used - Mattise. This was because the project is not actively mantained, slow to catch up on Android release updates (e.g. support AndroidX) etc. I had to fork it an mantain it myself, but given I am not an Android developer this was not very suitable.

So I've switched to use another Android image picker, which is more actively mantained, has exact same features as the previous library and allows easier customization in runtime, instead of writing xml files.

The case is that it does not support videos https://github.com/sangcomz/FishBun/issues/98

I am also not very convinced I would like to add support for videos now in this plugin, as it will be huge refactor and will cause a lot of breaking changes.

For now I will make sure it does properly what it should - picking images. When both native libraries that are used here support picking video I may reconsider adding such support.

Thanks again for your time and effort!
